src/runtime/SrcDocIFrameController.ts

```diff
@@ -142,10 +144,23 @@ export class SrcDocIFrameController extends Disposable implements IEmbeddedConte
         const controller = new IFrameController(this.host.contentWindow!.document.body);
 
         this.iframeController = controller;
-
-        this.dispose.add(controller.on('resize', (value) => this.updateIFrameHeight(value.height)));
         this.dispose.add(() => controller.dispose());
 
+        if (isNoScriptIFrame(this.host)) {
+            const noScriptIFrameController = new NoScriptIFrameController(this.host);
```

Comment:
a new controller is being created here, while the [old one](https://github.com/diplodoc-platform/html-extension/pull/61/files#diff-d9c685be8defd8defed9a65f6d59d66b6e3ead0ee9e6b2d87c6a26b6d221e66fR144) also exists, shouldn't this be done through conditional operators?

or if all class methods are required, then use inheritance

Reply:
I made this class as an addition to the main one, containing only code for working with iframes without scripts. But we can also try the option of moving the shared code to the base class and inheriting from it.

Reply:
This will not work, since NoScriptIFrameController depends on the iframe element itself, while IFrameController depends only on the inner body.
So this will require a more serious redesign of both classes.